Refactor: Weather 기능 수정 #27

dmdwn99 opened the pull request:
## 📕 제목
- #27 

## 📗 작업 내용

> 구현 내용 및 작업 했던 내역

- [x] 작업내용1: HourlyWeather, DailyWeather Entity 및 DTO 수정
- [x] 작업내용2: ErrorCode 500에서 404로 변경
- [x] 작업내용3: Weather Description 한글 변환을 위한 EnumType 구현
- [x] 작업내용4: WeatherService Json Parsing 수정
- [ ] 
## 📘 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

dmdwn99 committed:
Refactor: Weather 기능 수정 #27

1. HourlyWeather, DailyWeather Entity 및 DTO수정
2. ErrorCode 500에서 404로 변경
3. Weather Description 한글 변환을 위한 EnumType 구현

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/weather/dto/response/HourlyWeatherResponse.java:
```diff
@@ -12,13 +12,15 @@
                 .temp(hourlyWeather.getTemp())
                 .icon(hourlyWeather.getIcon())
+                .description(hourlyWeather.getDescription())
                 .regionName(hourlyWeather.getRegion().getName())
```
나중에 querydsl 같은 걸로 리팩토링합시다~!

reeruryu commented on src/main/java/com/seungah/todayclothes/domain/weather/service/WeatherService.java:
```diff
@@ -112,42 +118,45 @@ public void saveDailyWeather(String jsonString,Region region) throws ParseExcept
+                        dailyWeather.from(nightTemp, windSpeed, rain, humidity, icon, lowTemp, highTemp);
                         break;
                 }
                 dailyWeatherRepository.saveAll(dailyWeathers);
```
api 호출 부분이랑 getXXXWeather은 추후에 서비스 나누는 것도 좋을 것 같아요

dmdwn99 committed:
Refactor: Weather 기능 수정 #27

1. HourlyWeather, DailyWeather Entity 및 DTO수정
2. ErrorCode 500에서 404로 변경
3. Weather Description 한글 변환을 위한 EnumType 구현

dmdwn99 committed:
Refactor: Weather 기능 수정 #27

1. HourlyWeather, DailyWeather Entity 및 DTO수정
2. ErrorCode 500에서 404로 변경
3. Weather Description 한글 변환을 위한 EnumType 구현
4. 502 Bad Gateway를 방지하기 위한 MainController 구현
